app/uk/gov/hmrc/customs/declaration/controllers/actionbuilders/AuthActionCustomHeader.scala

```diff
@@ -75,14 +75,14 @@ class AuthActionSubmitterHeader @Inject()(customsAuthService: CustomsAuthService
 
   override def eitherCspAuthData[A](maybeNrsRetrievalData: Option[NrsRetrievalData])(implicit vhr: HasRequest[A] with HasConversationId): Either[ErrorResponse, AuthorisedAsCsp] = {
     val cpsAuth: Either[ErrorResponse, Csp] = for {
-      maybeBadgeId <- eitherBadgeIdentifier(allowNone = true).right
-      maybeEori <- eitherEori.right
+      maybeBadgeId <- eitherBadgeIdentifier(allowNone = true)
+      maybeEori <- eitherEori
     } yield {
       logger.info(headerValidator.logEoriAndBadgeIdHeaderText(XSubmitterIdentifierHeaderName, maybeBadgeId, maybeEori))
       Csp(maybeEori, maybeBadgeId, maybeNrsRetrievalData)
     }
 
-    if (cpsAuth.isRight && cpsAuth.right.get.badgeIdentifier.isEmpty && cpsAuth.right.get.eori.isEmpty) {
+    if (cpsAuth.isRight && cpsAuth.toOption.head.badgeIdentifier.isEmpty && cpsAuth.toOption.head.eori.isEmpty) {
```

Comment:
I think a clearer and more readable of expressing this bit of logic would be:

```scala
cpsAuth match {
  case Right(csp) if csp.badgeIdentifier.isEmpty && csp.eori.isEmpty =>
    logger.error(s"Both $XSubmitterIdentifierHeaderName and $XBadgeIdentifierHeaderName headers are missing")
    Left(errorResponseMissingIdentifiers)
  case _ =>
    cpsAuth
}
```

Generally, we should consider calls to `.isRight` and `.head` code smells as we can often use pattern matching and methods like `map` and `flatMap` instead.